### PR TITLE
chore: change ownership of profile sync from notifications to identity

### LINKED ISSUE #28900: [Extension] Update codebase + CODEOWNERS to reflect ownership change (notifications -> identity)

### PR DESCRIPTION
**Pull request by mathieuartu**
## **Description**

This PR takes care of all the necessary changes in order to decouple "Profile Sync" from the notifications feature.
This also underlines the ownership change from @MetaMask/notifications to @MetaMask/identity.

The changes made here will help transitioning to cleaner separation of concerns for features leveraging profile sync (i.e Notifications, Account syncing...).

⚠️ This PR does not add missing tests nor introduces any changes. This is only moving files around and separating concerns.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28901?quickstart=1)

## **Related issues**

Fixes https://github.com/MetaMask/metamask-extension/issues/28900

## **Manual testing steps**

No testing steps since this PR does not change the implementation of existing features.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -70,20 +70,23 @@ ui/selectors/institutional           @MetaMask/mmi
 ui/components/component-library      @MetaMask/design-system-engineers
 
 # The Notifications team is responsible for code related to notifications,
-# authentication, and profile syncing inside the Extension.
 
 # Controllers
-**/controllers/authentication/**                    @MetaMask/notifications
 **/controllers/metamask-notifications/**            @MetaMask/notifications
 **/controllers/push-platform-notifications/**       @MetaMask/notifications
-**/controllers/user-storage/**                      @MetaMask/notifications
```
Comment (mathieuartu) [inline .github/CODEOWNERS:77]: These two controllers have been moved to the external package `profile-sync-controller`

**Review by mirceanis — DISMISSED** [review]

**Review by Prithpal-Sooriya — DISMISSED** [review]
LGTM from notifications side.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -70,20 +70,23 @@ ui/selectors/institutional           @MetaMask/mmi
 ui/components/component-library      @MetaMask/design-system-engineers
 
 # The Notifications team is responsible for code related to notifications,
```
Comment (Gudahtt) [inline .github/CODEOWNERS:72]: Nit: Typo, trailing comma
```suggestion
# The Notifications team is responsible for code related to notifications
```

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Review by danjm — APPROVED** [review]
I am approving the codeowner file changes, and assume the other reviews cover the other file changes

**Pull request merged**